composer.json

```diff
@@ -5,7 +5,8 @@
     "description": "Apigee API Catalog for Drupal",
     "require": {
         "php": ">=7.1",
-        "drupal/apigee_edge": "~1.0-rc4"
+        "drupal/apigee_edge": "~1.0-rc4",
```

Comment:
Require as `^1.0` and add [conflict](https://getcomposer.org/doc/04-schema.md#conflict) as `<=1.0-rc4` that way you do not need to change the minimum requirement when 1.0 is out.

Reply:
This is simpler the way it is. Won't we want to require `apigee_edge` > 1.0.0 when it is released.

Reply:
I agree to leave it as is, will let user install anything above rc4, why force to 1.0.0?

Reply:
`^1.0` only enforces >=1.0.0 when 1.0 is out. Otherwise, it allows installation of any non-stable releases. The conflict solves that is always minimum `1.0-rc4`. This is the constraint that is being suggested on every Drupal.org module/theme install pages: https://www.drupal.org/project/apigee_edge/releases/8.x-1.0-rc4
> Install with Composer: $ composer require 'drupal/apigee_edge:^1.0'

**If 1.0.0 is out** but you left `~1.0-rc4` in this composer.json, depending on the [minimum-stability](https://getcomposer.org/doc/04-schema.md#minimum-stability) configuration of the Drupal project, it could happen that still `1.0-rc4` gets installed on the project. Although, if you use `^1.0' as a constraint then always an >=1.0.0 release gets installed.

Otherwise, if you leave `1.0-rc4` now and 1.0.0 is out and you would like to avoid the above-described scenario you will have to bump the minimum required version again to `^1.0` which could have been withdrawn by using what I am suggesting here.

I am fine if you merge this with the original constraint and later bump the minimum requirement later but sometimes it does not hurt if we think it through what a change means and how some extra work could be avoided.

Reply:
> Although, if you use `^1.0' as a constraint then always a >=1.0.0 release gets installed.

Well, I checked this in my lunchtime on http://semver.mwl.be and if it is not lying then it seems I expected more from Composer as I should have. It seems the `minimum-stability !== stable` can still allow installation of RC releases after the stable is out ;S

For me, it still makes sense to `require` the upcoming `^1.0` stable tag and the unstable minimum requirement (`<=rc4`) as a `conflict` because at least semantically it means for any reviewer that we prefer the stable release from the `apigee_edge` module and gives fewer credits to the unstable rc4 release that we wanna forget after stable is out. But this is my opinion only.